mantidproject/mantid
# Fix multi-period event data loading bug for LARMOR

Fixes #17787

Loading multiperiod event files for LARMOR was producing incorrect results. Only the first child workspace was rotated correctly. The other child workspaces were not rotated but translated. LARMOR decided if the workspaces should be moved or translated based on the run number. There was a bug in how this is extracted from the workspaces name is it cannot be found in the logs.

**To test:**

Find the test data here: \olympic\Babylon5\Scratch\Anton\LARMOR_MULTIPERIOD_BUG
1. Open the ISIS SANS GUI
2. Set the instrument to LARMOR
3. Load the User File : USER_Larmor_163F_HePATest_r13038.txt
4. Set the scatter sample to LARMOR00013038
5. Press Load
6. Run the script below

``` python
base_name = "13038_sans_nxs_"
for index in range(1, 5):
    ws_name = base_name + str(index)
    ws = mtd.retrieve(ws_name)
    instrument = ws.getInstrument()
    component =  instrument.getComponentByName("DetectorBench")
    print "Workspace {0} has a position of {1} and rotation of {2}".format(index,  component.getPos(), component.getRotation())
```

Confirm that all positions and rotations are the same.


## Relase notes can be found [here](https://github.com/mantidproject/mantid/pull/17796/commits/1323560d01abc5364e988e4bb7816d0c41f51b2a)
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
